### ISSUE TITLE
Unused variable does not take string interpolation into account

### ISSUE THREAD

**Issue by levibotelho** [palantir/tslint#1291]
### Bug Report

- __TSLint version__: `3.10.2`
- __TypeScript version__: `1.8.10`
- __Running TSLint via__: gulp-tslint

#### TypeScript code being linted

```ts
const foo = "bar";
const baz = `qux{foo}`;
console.log(baz);
```

#### Actual behavior

[gulp-tslint] error (no-unused-variable) weather.component.ts[12, 15]: unused variable: 'foo'

#### Expected behavior

There shouldn't be any error because `foo` is being used.

**Comment by adidahiya** [comment]
Is there a typo in your code sample? That's not the correct syntax for string interpolation; you need a `$`:

```ts
const baz = `qux${foo}`;
```

**Comment by levibotelho** [comment]
Ugh typo you're absolutely correct, sorry for the bother :).
